### PR TITLE
improve install_kata_kernel.sh script

### LINKED ISSUE #4453: experimental and vanilla kernels are the same kernel
http://jenkins.katacontainers.io/job/kata-containers-2.0-kernel-experimental-x86_64-nightly/ and http://jenkins.katacontainers.io/job/kata-containers-2.0-kernel-vanilla-x86_64-nightly/ are basically the same kernel, is this expected?

see https://github.com/kata-containers/kata-containers/issues/3119

### LINKED ISSUE #4454: ci: install_kata_kernel.sh is not working properly
./install_kata_kernel.sh downloads and installs the cache kernel even when the release version is different

```
$ ./install_kata_kernel.sh 
INFO: Kata guest kernel : 5.10.25-88
INFO: cached kernel  : 5.10.25-87
INFO: Attempting to download cached vmlinux
########################################################################################################################################### 100.0%
INFO: Attempting to download cached vmlinuz
########################################################################################################################################### 100.0%
vmlinux-5.10.25-87: OK
vmlinuz-5.10.25-87: OK
INFO: Installed pre-built kernel
```

### PR DESCRIPTION
**Pull request by devimc**
fixes #4453
fixes #4454

### CODE REVIEW

**Review by fidencio — DISMISSED** [review]
lgtm, @devimc!

@wainersm, it reminds me, do we check somewhere, anywhere, for the architecture of the cached kernel? Or do we have the possibility of trying to install a cached x86_64 kernel on, let's say. a pcc64le machine?

**Review by GabyCT — DISMISSED** [review]

**Comment by devimc** [comment]
@fidencio @GabyCT I have added a workaround for https://github.com/kata-containers/kata-containers/issues/3145 bug

/test

File: .ci/install_kata_kernel.sh [hunk 1]
```diff
@@ -66,20 +86,89 @@ install_prebuilt_kernel() {
...
+	Build and install a kernel for Kata Containers
+
+Usage:
+
+	$script_name [options]
+
+Options:
+    -d          : Enable bash debug.
+    -h          : Display this help.
+    -t <kernel> : kernel type, such as vanilla, experimental, tdx.
```
Comment (jodh-intel) [inline .ci/install_kata_kernel.sh:132]: Idea: It's a bit more work, but you could consider adding a `-l` option to list the available types by using a hash of `type=>key:description`. Something like:

```bash
 kernel_types=(
    [experimental]="assets.kernel-experimental.tag:Experimental kernel"
    [tdx]="assets.kernel.tdx.tag:an Intel TDX (Trust Domain Extensions) kernel"

    # ...
  )
```

File: .ci/install_kata_kernel.sh [hunk 2]
```diff
@@ -66,20 +86,89 @@ install_prebuilt_kernel() {
...
+	clone_katacontainers_repo
+
+	case "${kernel_type}" in
+		vanilla)
+			install_kernel "${kernel_type}" $(get_version "assets.kernel.version")
+			;;
+		experimental)
+			install_kernel "${kernel_type}" $(get_version "assets.kernel-experimental.tag")
+			;;
+		tdx)
```
Comment (jodh-intel) [inline .ci/install_kata_kernel.sh:164]: Nit: Unsorted list.

**Review by jodh-intel — APPROVED** [review]
Thanks @devimc.

**Review by fidencio — APPROVED** [review]
lgtm, again, thanks @devimc! :-)

**Review by GabyCT — APPROVED** [review]

**Pull request merged**